internal/pkg/cli/deploy/workload.go

```diff
@@ -302,27 +314,62 @@ func (d *workloadDeployer) forceDeploy(in *forceDeployInput) error {
 	return nil
 }
 
-func (d *workloadDeployer) uploadContainerImage(imgBuilderPusher imageBuilderPusher) (*string, error) {
+func (d *workloadDeployer) uploadContainerImage(imgBuilderPusher imageBuilderPusher) (*string, map[string]string, error) {
```

Comment:
What about ⬇️  skeleton?
```go
// In manifest package
func DockerfileBuildRequired() (bool, error) {
   // Return true if >= 1 container (main + sidecars) needs docker build.
   // Return false otherwise.
}

// here
func ((d *workloadDeployer) uploadContainerImage(imgBuilderPusher imageBuilderPusher) (*string, map[string]string, error) {
  required, err := manifest.DockerBuildRequired(d.mft)
  if !required { return nil, nil }
  ...
}
```

Reply:
Can we simplify and have have the following signature?
```suggestion
func (d *workloadDeployer) uploadContainerImage(imgBuilderPusher imageBuilderPusher) (map[string]string, error) {
```
So that the image digest for the main container is also included in the map?